src/nfa/complement.cc

```diff
@@ -8,43 +8,28 @@
 using namespace mata::nfa;
 using namespace mata::utils;
 
-Nfa mata::nfa::algorithms::complement_classical(const Nfa& aut, const OrdVector<Symbol>& symbols,
-                                                bool minimize_during_determinization) {
-    Nfa result;
-    State sink_state;
-    if (minimize_during_determinization) {
-        result = minimize_brzozowski(aut); // brzozowski minimization makes it deterministic
-        if (result.final.empty() && !result.initial.empty()) {
-            assert(result.initial.size() == 1);
-            // if automaton does not accept anything, then there is only one (initial) state
-            // which can be the sink state (so we do not create unnecessary one)
-            sink_state = *result.initial.begin();
-        } else {
-            sink_state = result.num_of_states();
-        }
-    } else {
-        std::unordered_map<StateSet, State> subset_map;
-        result = determinize(aut, &subset_map);
-        // check if a sink state was not created during determinization
-        auto sink_state_iter = subset_map.find({});
-        if (sink_state_iter != subset_map.end()) {
-            sink_state = sink_state_iter->second;
-        } else {
-            sink_state = result.num_of_states();
-        }
-    }
+Nfa mata::nfa::algorithms::complement_classical(const Nfa& aut, const OrdVector<Symbol>& symbols) {
+    return determinize(aut)
+        .trim()
+        .complement_deterministic(symbols);
```

Comment:
Do we want to remove the check if a sink state does not already exists from determinization? It probably makes sense, I'm just interested about your reasoning.

Reply:
There was a discussion about it in #342. Since we are trimming the determinized automaton, there will be no sink state anymore. The check would make sense if we did not perform the trim operation beforehand. Therefore, the options are:
1. trim and add a new sink state always, or
2. do not trim and then check for a sink state.

However, the second purpose of this rewrite was that the implementation of the complement is not dependent on the data structures and algorithm used for determinization, as per #342. If we were to keep the check without trimming, we would either have to keep the dependency on the subset_map, or find a sink state ourselves.

Personally, I would prefer to keep the original dependency on the subset map to find the sink state, but I believe that `{}` used for the sink state is no longer being created by the determinization algorithm.